src/AgDatabase.cs

```diff
@@ -50,7 +50,15 @@ public AgDatabase(DatabaseConfig dbConfig)
     {
       Name = dbConfig.DatabaseName;
       _listener = new Listener(new SqlConnectionStringBuilder(dbConfig.ConnectionString) { InitialCatalog = "master" });
-      _backupPathTemplate = dbConfig.BackupPathTemplate;
+
+      if(!string.IsNullOrWhiteSpace(dbConfig.BackupPathSqlQuery))
+        using(var cmd = _listener.Primary.SqlConnection.CreateCommand()) {
+          cmd.CommandText = dbConfig.BackupPathSqlQuery;
+          using(var reader = cmd.ExecuteReader()) {
+            if(reader.Read())
+              _backupPathTemplate = (string)reader[0];
+          }
+        }

```

Comment:
Does this resolution belong at the AgDatabase level or should we do this at the server level since it could be different for each host? I lean towards leaving the resolution to the point where we take the backup as it prepares us for better handling of fail over scenarios and retries.

Reply:
Yeah, I can push the implementation down to `Server`.
re - "it could be different for each host": How do you want to handle that?  I was only going to use the Primary, which is what is being done [here](https://github.com/factset/AgDatabaseMove/blob/master/src/AgDatabase.cs#L91)

Reply:
I don't think our current implementation handles a primary change during the AgDatabase object lifetime. Concerns about it working today on multiple hosts isn't at issue. It's just _if_ we made changes to support the [primary changing with fail over](https://github.com/factset/AgDatabaseMove/blob/master/src/SmoFacade/Listener.cs#L83) this means less state to manage or refactor out.

Reply:
I made these changes in the latest commit